packages/realm/src/app-services/App.ts

```diff
@@ -206,46 +209,80 @@ export class App<FunctionsFactoryType = DefaultFunctionsFactory, CustomDataType
     return this.internal.config.appId;
   }
 
+  /**
+   * Log in a user.
```

Comment:
```suggestion
   * Log in a user.
   * @param credentials - {@link Credentials} object which determines the authentication method.
   * @returns {@link User}
```

Reply:
I also think we should explicitly define return types on our public methods (not just jsdoc and implicit).  Makes it less likely that users will have to dig through our code to figure out what's going on.

Reply:
It's a recommend setting to leave type information out of tsdoc comments as it's considered redundant with the TS type declarations.

Reply:
Sorry, my comment isn't clear.  This method (`logIn`) in particular is not showing a return type in the TS type declarations.  I believe it returns a `Promise<User>`

Reply:
> It's a recommend setting to leave type information out of tsdoc comments as it's considered redundant with the TS type declarations.

I'm fine with that if it's clear from the code.